test/server/config_validation/xds_verifier.h

```diff
@@ -0,0 +1,76 @@
+#include "envoy/admin/v3/config_dump.pb.h"
+#include "envoy/common/exception.h"
+#include "envoy/config/cluster/v3/cluster.pb.h"
+#include "envoy/config/endpoint/v3/endpoint.pb.h"
+#include "envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.pb.h"
+#include "envoy/config/listener/v3/listener.pb.h"
+#include "envoy/config/route/v3/route.pb.h"
+
+#include "common/common/assert.h"
+
+#include "test/server/config_validation/xds_fuzz.pb.h"
+
+#include "absl/container/flat_hash_map.h"
+
+namespace Envoy {
+
+class XdsVerifier {
+public:
+  XdsVerifier(test::server::config_validation::Config::SotwOrDelta sotw_or_delta);
+  void listenerAdded(const envoy::config::listener::v3::Listener& listener,
+                     bool from_update = false);
+  void listenerUpdated(const envoy::config::listener::v3::Listener& listener);
+  void listenerRemoved(const std::string& name);
+  void drainedListener(const std::string& name);
+
+  void routeAdded(const envoy::config::route::v3::RouteConfiguration& route);
+  void routeUpdated(const envoy::config::route::v3::RouteConfiguration& route);
+
+  enum ListenerState { WARMING, ACTIVE, DRAINING, REMOVED };
+  struct ListenerRepresentation {
+    envoy::config::listener::v3::Listener listener;
+    ListenerState state;
+  };
+
+  const std::vector<ListenerRepresentation>& listeners() const { return listeners_; }
+
+  uint32_t numWarming() { return num_warming_; }
```

Comment:
Nit: all these accessor methods can be const, i.e. `uint32_t numWarming() const { ...`